utils/python/CIME/buildlib.py

```diff
@@ -0,0 +1,49 @@
+from CIME.XML.standard_module_setup import *
+from CIME.case import Case
+"""
+buildlibx - given the caseroot and component name(cmp), build the stub component
+"""
+
+class buildlib(object):
```

Comment:
In standard python convention, class names should be capitalized in order to look different than functions. Please change this to BuildLib and make similar changes to the other three classes you added.